canonicalwebteam/blog/common_view_logic.py

```diff
@@ -5,40 +5,151 @@
+        if category:
+            category_resolved = api.get_category_by_slug(category)
+            category_id = category_resolved["id"]
+
+        upcoming = []
+        featured_articles = []
+        featured_article_ids = []
+        if page == "1":
+            featured_articles, total_pages = api.get_articles(
+                tags=self.tag_ids,
+                tags_exclude=self.excluded_tags,
+                page=page,
+                sticky="true",
+                per_page=3,
+            )
+
+            featured_article_ids = [
+                article["id"] for article in featured_articles
+            ]
+
+            if enable_upcoming:
+                # Maybe we can get the IDs since there is no chance
+                # this going to move
+                events = api.get_category_by_slug("events")
+                webinars = api.get_category_by_slug("webinars")
+                upcoming, _ = api.get_articles(
+                    tags=self.tag_ids,
+                    tags_exclude=self.excluded_tags,
+                    page=page,
+                    per_page=3,
+                    categories=[events["id"], webinars["id"]],
+                )
+
+        articles, total_pages = api.get_articles(
+            tags=self.tag_ids,
+            tags_exclude=self.excluded_tags,
+            exclude=featured_article_ids,
+            page=page,
+            categories=[category_id],
+        )
+
+        context = get_index_context(
+            page,
+            articles,
+            total_pages,
+            featured_articles=featured_articles,
+            upcoming=upcoming,
+        )
+
+        context["title"] = self.blog_title
+        context["category"] = {"slug": category}
+        context["upcoming"] = upcoming
+
+        return context
+
+
+def get_embedded_categories(embedded):
+    """Returns the categories in the embedded response from wp
+    The category is in the first object of the wp:term of the response:
+    embedded["wp:term"][0]
+
+
+    :param embedded: The embedded dictionnary in teh response
+    :returns: Dictionnary of categories
+    """
+    if "wp:term" in embedded and embedded["wp:term"][0]:
+        return embedded["wp:term"][0]
+    return {}
+
+
+def get_embedded_group(embedded):
+    """Returns the group in the embedded response from wp.
+    The group is in the fourth object of the wp:term list of the response:
+    embedded["wp:term"][3]
+
+
+    :param embedded: The embedded dictionnary in the response
+    :returns: Dictionnary of group
+    """
+    if "wp:term" in embedded and embedded["wp:term"][3]:
+        return embedded["wp:term"][3][0]
+    return {}
+
+
+def get_embedded_author(embedded):
+    """Returns the author in the embedded response from wp.
+    embedded["author"]
+
+
+    :param embedded: The embedded dictionnary in the response
+    :returns: Dictionnary of author
+    """
+    if "author" in embedded:
+        return embedded["author"][0]
+    return {}
```

Comment:
return embedded.get("author", [{}])[0]?

Reply:
I am not a big fan of calling the element of a list directly after a function call.

Reply:
```
    authors = embedded.get("author", [{}])
    return authors[0]
```

?